### PR TITLE
[Docs] fix Types of transactions

### PR DESCRIPTION
**Pull request by nkysg**
### Description
add module bundle transaction
### Test Plan
n/a

### CODE REVIEW

**Comment by clay-aptos** [comment]
Thank you for the contribution! I am copying @davidiw as primary author and @gregnazario as resident expert to confirm this addition makes sense.

Also, I see the inconsistency between the SDK support statements here and am asking David and Greg if I can reconcile or perhaps even single source (combine) those sections and then link from the Systems Integrators Guide to the Transactions concept for transaction types.

David, what do you think? Thanks again!

File: developer-docs-site/docs/concepts/txns-states.md [hunk 1]
```diff
@@ -59,9 +59,10 @@ A [signed transaction](../guides/sign-a-transaction.md) on the blockchain contai
 - **Expiration time**: A timestamp after which the transaction ceases to be valid (i.e., expires).
 
 ### Types of transactions
-Within a given transaction, the target of execution can be one of two types:
+Within a given transaction, the target of execution can be one of three types:
 
 - An entry point
+- module bundle (publishes multiple modules)
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/txns-states.md:65]: Module bundle is deprecated, and does not work, which is why it was removed.

Instead, module code is updated via an entry function via a package

**Comment by gregnazario** [comment]
> Thank you for the contribution! I am copying @davidiw as primary author and @gregnazario as resident expert to confirm this addition makes sense.
> 
> Also, I see the inconsistency between the SDK support statements here and am asking David and Greg if I can reconcile or perhaps even single source (combine) those sections and then link from the Systems Integrators Guide to the Transactions concept for transaction types.
> 
> David, what do you think? Thanks again!

This is actually something that's deprecated.  Module bundle transaction was replaced with the package system that doesn't use a custom transaction type for it.

**Comment by gregnazario** [comment]
Thanks for your contribution; however, it no longer applies since the type is deprecated.  I'm unsure if we would want to just put deprecated, or leave it as only 2 types (as was mentioned previously)

File: developer-docs-site/docs/concepts/txns-states.md (hunk 1 above)
Comment (nkysg) [reply developer-docs-site/docs/concepts/txns-states.md:65]: thanks, I will close this pr.

**Comment by nkysg** [comment]
> Thanks for your contribution; however, it no longer applies since the type is deprecated. I'm unsure if we would want to just put deprecated, or leave it as only 2 types (as was mentioned previously)

Thanks for your reply. Maybe you can remove ModuleBundle type from code. It's convenient for people who want to read aptos code.

**Comment by clay-aptos** [comment]
> > Thank you for the contribution! I am copying @davidiw as primary author and @gregnazario as resident expert to confirm this addition makes sense.
> > Also, I see the inconsistency between the SDK support statements here and am asking David and Greg if I can reconcile or perhaps even single source (combine) those sections and then link from the Systems Integrators Guide to the Transactions concept for transaction types.
> > David, what do you think? Thanks again!
> 
> This is actually something that's deprecated. Module bundle transaction was replaced with the package system that doesn't use a custom transaction type for it.

I was thinking these:
"This guide points out many of those entry points, such as `coin::transfer` and `aptos_account::create_account`."
"This guide points out many of those entry points, such as `aptos_account::transfer` and `aptos_account::create_account`."

Perhaps this is for @movekevin to determine? `coin::transfer` or `aptos_account::transfer`? Per:
https://github.com/aptos-labs/aptos-core/pull/6835

**Comment by movekevin** [comment]
> "This guide points out many of those entry points, such as `aptos_account::transfer` and `aptos_account::create_account`."

Currently aptos_account::transfer is only for APT and it can implicitly create the recipient account if it doesn't exist.
In about 1-2 weeks, we'll have aptos_account::transfer_coins in mainnet, which for both APT and non-APT coins, implicitly creates the recipient account if doesn't exist and implicitly register the recipient account to receive the CoinType if not registered before.

**Review by davidiw — CHANGES_REQUESTED** [review]
if you want this, you could say it was deprecated. Also there's a new multi-account @movekevin

**Pull request closed without merging**